Address linter warnings

```console
$ golangci-lint run ./pkg/...

pkg/sources/secret/secret.go:63:49: S1019: should use make(map[string]*corev1.Secret) instead (gosimple)
        secretCache := make(map[string]*corev1.Secret, 0)
                                                       ^
pkg/transformation/pipeline/handler.go:62:19: SA1019: cloudevents.NewDefaultClient is deprecated: Please use NewClientHTTP with the observability options. (staticcheck)
        ceClient, err := cloudevents.NewDefaultClient()
                         ^
pkg/routing/adapter/splitter/adapter.go:168:16: Error return value of `newCE.SetData` is not checked (errcheck)
                newCE.SetData(cloudevents.ApplicationJSON, v.Raw)
                             ^
```

--- a/pkg/routing/adapter/splitter/adapter.go
+++ b/pkg/routing/adapter/splitter/adapter.go
@@ -165,7 +165,10 @@ func (h *Handler) split(path string, e *event.Event) []*event.Event {
 	}
 	for _, v := range val.Array() {
 		newCE := cloudevents.NewEvent()
-		newCE.SetData(cloudevents.ApplicationJSON, v.Raw)
+		if err := newCE.SetData(cloudevents.ApplicationJSON, v.Raw); err != nil {
+			h.logger.Errorw("Failed to set event data", zap.Error(err))
+			continue
+		}
 		result = append(result, &newCE)
 	}
 	return result

--- a/pkg/sources/secret/secret.go
+++ b/pkg/sources/secret/secret.go
@@ -60,7 +60,7 @@ func (g *GetterWithClientset) Get(refs ...v1alpha1.ValueFromField) (Secrets, err
 
 	// cache Secret objects by name between iterations to avoid multiple
 	// round trips to the Kubernetes API for the same Secret object.
-	secretCache := make(map[string]*corev1.Secret, 0)
+	secretCache := make(map[string]*corev1.Secret)
 
 	for _, ref := range refs {
 		val := ref.Value

--- a/pkg/transformation/pipeline/handler.go
+++ b/pkg/transformation/pipeline/handler.go
@@ -59,7 +59,7 @@ func NewHandler(context, data []v1alpha1.Transform) (Handler, error) {
 	contextPipeline.setStorage(sharedVars)
 	dataPipeline.setStorage(sharedVars)
 
-	ceClient, err := cloudevents.NewDefaultClient()
+	ceClient, err := cloudevents.NewClientHTTP()
 	if err != nil {
 		return Handler{}, err
 	}

